runtime/master/src/main/java/org/apache/nemo/runtime/master/resource/ContainerManager.java

```diff
@@ -191,11 +194,23 @@ public void onContainerAllocated(final String executorId,
       messageSender = new FailedMessageSender();
     }
 
+    final ExecutorRepresenter executorRepresenter;
     // Create the executor representation.
-    final ExecutorRepresenter executorRepresenter =
-      new ExecutorRepresenter(executorId, resourceSpec, messageSender, activeContext, serializationExecutorService,
-        activeContext.getEvaluatorDescriptor().getNodeDescriptor().getName());
-
+    if(this.executorType.equals("default")) {
```

Comment:
why don't you use Factory pattern? 
It would be good to create an interface: `ExecutorRepresenterFactory`  
and two implementation factories: `DefaultExecutorRepresenterFactory` and `LambdaExecutorRepresenterFactory`. 

Also we can inject the specific factory through Tang and don't have to use `if--else` here.